File: reference/array-hash/128-Longest-Consecutive-Sequence.go
func longestConsecutive(nums []int) int {
	set := make(map[int]bool)
	for _, num := range nums {
		set[num] = true
	}

	maxlen := 0
	for num := range set {
		if !set[num-1] {
			length := 1
			current := num

			for set[current+1] {
				current += 1
				length++
			}

			if length > maxlen {
				maxlen = length
			}
		}
	}
	return maxlen
}

File: reference/array-hash/128-Longest-Consecutive-Sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        numSet = set(nums)
        maxLength = 0

        for num in numSet:
            if num - 1 not in numSet:
                current = 0
                while num + current in numSet:
                    current += 1
                maxLength = max(maxLength, current)
        return maxLength